src/site/content/en/fast/lazy-loading-problems/index.md

```diff
@@ -0,0 +1,199 @@
+---
+layout: post
+title: Lazy-loading problems
+authors:
+  - jeremywagner
+  - rachelandrew
+date: 2019-08-16
+updated: 2020-06-11
+description: |
+  This post details some of the potential pitfalls of lazy-loading.
+tags:
+  - performance
+---
+
+While lazy-loading images and video have positive and measurable performance
+benefits, it's not a task to be taken lightly. If you get it wrong, there could
+be unintended consequences. As such, it's important to keep the following
+concerns in mind:
+
+## Mind the fold {: #wrong-fold }
+
+It may be tempting to lazy-load every single media resource on the page with
+JavaScript, but you need to resist this temptation. Anything resting above the
+fold shouldn't be lazy-loaded. Such resources should be considered critical
+assets, and thus should be loaded normally.
+
+The primary argument for loading critical media resources the usual way in lieu
+of lazy-loading is that lazy-loading delays the loading of those resources until
+after the DOM is interactive when scripts have finished loading and begin
+execution. For images below the fold, this is fine, but it would be faster to
+load critical resources above the fold with a standard `<img>` element.
+
+Of course, where the fold lies is not so clear these days when websites are
+viewed on so many screens of varying sizes. What lies above the fold on a laptop
+may well lie _below_ it on mobile devices. There's no bulletproof advice for
+addressing this optimally in every situation. You'll need to conduct an
+inventory of your page's critical assets, and load those images in typical
+fashion.
+
+Additionally, you may not want to be so strict about the fold line as the
+threshold for triggering lazy-loading. It may be more ideal for your purposes to
+establish a buffer zone some distance below the fold so that images begin
+loading well before the user scrolls them into the viewport. For example, The
+Intersection Observer API allows you to specify a `rootMargin` property in an
+options object when you create a new `IntersectionObserver` instance. This
+effectively gives elements a buffer, which triggers lazy-loading behavior before
+the element is in the viewport:
+
+```javascript
+let lazyImageObserver = new IntersectionObserver(function(entries, observer) {
+  // lazy-loading image code goes here
+}, {
+  rootMargin: "0px 0px 256px 0px"
+});
+```
+
+If the value for `rootMargin` looks similar to values you'd specify for a CSS
+`margin` property, that's because it is! In this case, we're broadening the
+bottom margin of the observing element (the browser viewport by default, but
+this can be changed to a specific element using the `root` property) by 256
+pixels. That means the callback function will execute when an image element is
```

Comment:
```suggestion
If the value for `rootMargin` looks similar to values you'd specify for a CSS
`margin` property, that's because it is! In this case, the
bottom margin of the observed element (the browser viewport by default, but
this can be changed to a specific element using the `root` property) is broadened by 256
pixels. That means the callback function will execute when an image element is
```